File: src/main/scala/s/decorator/inheritance/SideBorder.scala
package s.decorator.inheritance

final class SideBorder(display: Display, private val borderChar: Char)
    extends Border(display) {
  override def columns: Int = 1 + display.columns + 1

  override def rows: Int = display.rows

  override def rowText(row: Int): String =
    borderChar.toString + display.rowText(row) + borderChar.toString
}

File: src/main/java/j/decorator/inheritance/SideBorder.java
package j.decorator.inheritance;

public final class SideBorder extends Border {
  private final char borderChar;

  SideBorder(Display display, char ch) {
    super(display);
    this.borderChar = ch;
  }

  @Override
  public int getColumns() {
    return 1 + display.getColumns() + 1;
  }

  @Override
  public int getRows() {
    return display.getRows();
  }

  @Override
  public String getRowText(int row) {
    return borderChar + display.getRowText(row) + borderChar;
  }
}
